### PR TITLE
Added test for accessor methods in flobject.py

### PR DESCRIPTION
**Pull request by prmukherj**
This tests the availability / behavior of accessor methods like: is_active, is_read_only, min, max, default_value and allowed_values, on settings objects of all types.

### CODE REVIEW

**Comment by prmukherj** [comment]
@seanpearsonuk, this PR is ready. This test is taking approximately 85 seconds to run. Shall we merge this to the main branch or keep the test only as art of nightly CI?

File: tests/test_flobject.py [hunk 1]
```diff
@@ -749,3 +749,74 @@ def test_settings_matching_names(new_solver_session_no_transcript) -> None:
...
+        if type_data == "Boolean":
+            assert {
+                "is_active",
+                "is_read_only",
+                "default_value",
+                "get_state",
+                "set_state",
+            }.issubset(set(dir(nodes[type_data])))
+            assert not nodes[type_data].is_read_only()
+            assert nodes[type_data].is_active()
```
[the remark this answers is not in the document]
Comment (prmukherj) [reply tests/test_flobject.py:845]: Done, Thank you.

File: tests/test_flobject.py [hunk 2]
```diff
@@ -749,3 +749,74 @@ def test_settings_matching_names(new_solver_session_no_transcript) -> None:
...
+
+def get_child_nodes(node, nodes, type_list):
+    if isinstance(node, flobject.NamedObject):
+        for item in node.get_object_names():
+            get_child_nodes(node[item], nodes, type_list)
+    elif isinstance(node, flobject.Group):
+        for item in node.get_active_child_names():
+            get_child_nodes(getattr(node, item), nodes, type_list)
+    else:
+        node_type = str(node.__class__.__bases__).split(".")[-1][:-4]
```
[the remark this answers is not in the document]
Comment (prmukherj) [reply tests/test_flobject.py:890]: Done, Thank you

File: tests/test_flobject.py [hunk 3]
```diff
@@ -749,3 +749,74 @@ def test_settings_matching_names(new_solver_session_no_transcript) -> None:
...
+            assert not nodes[type_data].is_read_only()
+            assert nodes[type_data].is_active()
+
+
+def get_child_nodes(node, nodes, type_list):
+    if isinstance(node, flobject.NamedObject):
+        for item in node.get_object_names():
+            get_child_nodes(node[item], nodes, type_list)
+    elif isinstance(node, flobject.Group):
+        for item in node.get_active_child_names():
```
[the remark this answers is not in the document]
Comment (prmukherj) [reply tests/test_flobject.py]: Implemented it, could you please have a quick look? Thank you @mkundu1 .

**Review by mkundu1 — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

File: tests/test_flobject.py [hunk 4]
```diff
@@ -820,3 +820,92 @@ def test_settings_matching_names(new_solver_session_no_transcript) -> None:
...
+            for item in node.child_names:
+                get_child_nodes(
+                    getattr(node, item), nodes, type_list, expected_type_list
+                )
+        else:
+            node_type = node.__class__.__bases__[0].__name__
+            if node_type not in type_list:
+                type_list.append(node_type)
+                nodes[node_type] = node
+                if type_list.sort() == expected_type_list.sort():
```
Comment (mkundu1) [inline tests/test_flobject.py:910]:
```
type_list = expected_type_list.copy()  # at the beginning

type_list.remove(node_type)
if not type_list():
  return
```

File: tests/test_flobject.py [hunk 5]
```diff
@@ -820,3 +820,92 @@ def test_settings_matching_names(new_solver_session_no_transcript) -> None:
...
+            assert {"is_active", "is_read_only", "get_state", "set_state"}.issubset(
+                set(dir(nodes[type_data]))
+            )
+            assert nodes[type_data].is_read_only() in [True, False]
+            assert nodes[type_data].is_active() in [True, False]
+
+
+def get_child_nodes(node, nodes, type_list, expected_type_list):
+    if node.is_active():
+        if isinstance(node, flobject.NamedObject):
```
Comment (mkundu1) [inline tests/test_flobject.py:897]: we can skip this as no case is read

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**